src/parser/DatasetClauses.h

```diff
@@ -7,15 +7,14 @@
 
 #include <vector>
 
-#include "index/ScanSpecification.h"
 #include "parser/sparqlParser/DatasetClause.h"
 
 namespace parsedQuery {
 
 // A struct for the FROM [NAMED] clause in queries, and the `USING [NAMED]` and
 // `WITH` clauses from `SPARQL Update`.
 struct DatasetClauses {
-  using Graphs = ScanSpecificationAsTripleComponent::Graphs;
+  using Graphs = std::optional<ad_utility::HashSet<TripleComponent>>;
```

Comment:
Why can't those be the `GraphFilter`, and the default for `defaultGraphs` is `all`, and the default for `namedGraphs` is `Blacklist{defaultGraph}` , then you can get rid of several of the remaining conversion functions in the query planner I guess.